docs/proof-trace.md

```diff
@@ -49,6 +49,9 @@ name              ::= string
 location          ::= string
 function          ::= WORD(0xDD) name location arg* WORD(0x11) // the arg list is ommited in normal mode
 
+function_name     ::= string
+pattern_matching_failure ::= WORD(0x44) function_name
```

Comment:
This needs to appear in the right hand side of some production, most likely the `argument` and `event` productions.